apps/tasks/views.py

```diff
@@ -82,23 +112,174 @@ def get(self, request, pk):
+
+
+class StopTaskView(UpdateAPIView):
+    serializer_class = Serializer
+    queryset = TimeLog.objects.all()
+
+    @serialize_decorator(TimeLogStopSerializer)
+    @swagger_auto_schema(request_body=no_body)
+    def patch(self, request, pk):
+        # TODO: add check current user
+        timelog = TimeLog.objects.filter(task_id=pk).filter(stop_time__isnull=True).first()
+        if not timelog:
+            return Response({"stopped": False, "cause": "Already stopped or not found!"})
+
+        timelog.stop_time = timezone.now()
+        timelog.duration = timelog.stop_time - timelog.start_time
+        timelog.save(update_fields=['stop_time', 'duration'])
+
+        return Response({"stopped": True})
+
+
+class AddTimelogView(CreateAPIView):
+    serializer_class = TimeLogManualSerializer
+    queryset = TimeLog.objects.all()
+
+    @serialize_decorator(TimeLogManualSerializer)
+    def post(self, request, pk):
+        validated_data = request.serializer.validated_data
+
+        task_id = self.kwargs['pk']
+        current_user = self.request.user
+        qs = Task.objects.filter(user=current_user).filter(id=task_id).filter(user_id=self.request.user.id)
+        if not len(qs):
+            return Response({"created": False, "cause": "Task is not owned by current user!"})
+
+        current_time = timezone.now()
+        timelog = TimeLog.objects.create(
+            start_time=current_time,
+            duration=validated_data['duration'] * 60,
+            stop_time=current_time + validated_data['duration'] * 60,
+            task_id=pk
+        )
+
+        return Response(TimeLogSerializer(timelog).data)
+
+
+class LastMonthTimelogView(ListAPIView):
+    serializer_class = TimeLogSerializer
+    queryset = TimeLog.objects.all()
+
+    def get(self, request, *args, **kwargs):
+        start_month_datetime = get_start_month_datetime()
+        total_time = TimeLog.objects.filter(task__user_id=self.request.user.id).filter(
+            start_time__gt=start_month_datetime) \
+            .aggregate(total_time=Sum('duration'))
+        print(total_time['total_time'])
+        return Response({"total_time": total_time['total_time'].seconds})
+
+
+class LastMonthTopTimelogView(ListAPIView):
+    serializer_class = TaskSerializer
+    queryset = Task.objects.all()
+
+    def get(self, request, *args, **kwargs):
+        start_month_datetime = get_start_month_datetime()
+        tasks = Task.objects.filter(user=self.request.user).filter(
+            timelog__start_time__gt=start_month_datetime) \
+            .values('id', 'title').annotate(total_time=Sum('timelog__duration')) \
+            .order_by('total_time')[:20]
+        return Response(tasks)
+
+
+class GetTimelogView(ListAPIView):
+    serializer_class = TimeLogSerializer
+
+    def get_queryset(self):
+        task_id = self.kwargs.get('pk', 0)
+        return TimeLog.objects.filter(task_id=task_id)
+
+
+def send_notification(notification_type, task):
```

Comment:
functiile ajutatoare nu trebuiesc puse in views.py, creaza un fisier helpers.py si le tii acolo